pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -473,6 +481,17 @@ else let
                    else true);
     };
 
+  checkedEnv =
+    let
+      overlappingNames = lib.intersectLists (lib.attrNames env) (lib.attrNames derivationArg);
```

Comment:
lib.intersectLists has quadratic runtime behavior. Will that be a problem?

Reply:
We should probably use `builtins.attrNames (builtins.intersectAttrs env derivationArg)` instead.

Reply:
Or maybe change the implementation of `intersectLists` to do that by default?

Reply:
`intersectLists` has to work with any type of elements, not just strings (or strings-that-can-be-attribute-names, which I'm not sure whether that's a different type)

Reply:
It looks like there's a function `unionOfDisjoint` in `lib/attrsets.nix` that's not currently exposed. We could probably just use that.

Reply:
https://github.com/NixOS/nixpkgs/pull/206525